### PR TITLE
Suggest publish for running on an isolated machine

### PR DESCRIPTION
**Pull request by singhsarab**
## Description
For test projects targetting net core, give actionable message to the user in case test host file is not found.

## Related issue
https://github.com/Microsoft/vstest/issues/1585

### CODE REVIEW

File: src/Microsoft.TestPlatform.TestHostProvider/Resources/Resources.Designer.cs [hunk 1]
```diff
@@ -72,11 +72,20 @@ internal class Resources {
...
-        internal static string NoTestHostFileExist {
+        internal static string UnableToFindDepsFile {
             get {
-                return ResourceManager.GetString("NoTestHostFileExist", resourceCulture);
+                return ResourceManager.GetString("UnableToFindDepsFile", resourceCulture);
+            }
+        }
+
+        /// <summary>
+        ///   Looks up a localized string similar to Unable to find test host at {0}. Please publish your test project and retry.
```
Comment (abhishkk) [inline src/Microsoft.TestPlatform.TestHostProvider/Resources/Resources.Designer.cs]: Nit: comment should match with resource string

File: test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DotnetTestHostManagerTests.cs [hunk 2]
```diff
@@ -205,6 +205,27 @@ public void GetTestHostProcessStartInfoShouldIncludeEnvironmentVariables()
             Assert.AreEqual(environmentVariables, startInfo.EnvironmentVariables);
         }
 
+        [TestMethod]
+        public void GetTestHostProcessStartIfDepsFileAndTestHostNotFoundShouldThrowException()
+        {
+            this.mockFileHelper.Setup(fh => fh.Exists("test.deps.json")).Returns(false);
+            this.mockFileHelper.Setup(ph => ph.Exists("testhost.dll")).Returns(false);
```
Comment (abhishkk) [inline test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DotnetTestHostManagerTests.cs]: UT for testhost dll true and deps false case.

**Review by abhishkk — APPROVED** [review]
approved with suggestions

File: test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DotnetTestHostManagerTests.cs (hunk 2 above)
Comment (singhsarab) [reply test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DotnetTestHostManagerTests.cs]: Done

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs [hunk 3]
```diff
@@ -345,74 +336,90 @@ private string GetTestHostPath(string runtimeConfigDevPath, string depsFilePath,
...
 
-            // If we are here it means it couldnt resolve testhost.dll from nuget chache.
+            // If we are here it means it couldnt resolve testhost.dll from nuget cache.
             // Try resolving testhost from output directory of test project. This is required if user has published the test project
             // and is running tests in an isolated machine. A second scenario is self test: test platform unit tests take a project
             // dependency on testhost (instead of nuget dependency), this drops testhost to output path.
             testHostPath = Path.Combine(sourceDirectory, "testhost.dll");
             EqtTrace.Verbose("DotnetTestHostManager: Assume published test project, with test host path = {0}.", testHostPath);
 
+            if (!this.fileHelper.Exists(testHostPath))
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs]: seems like we could simply replace the message above at line 216 with this new resource?

Comment (singhsarab) [reply src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs]: We are caching the error above, and then throwing that.

**Comment by lextm** [comment]
The error message can be confusing, as at least for xUnit.net users, the right way is to install xunit VS adapter NuGet package, not publish the test project itself.

**Pull request merged**